content/blog/spring-cleaning/index.Rmd

```diff
@@ -0,0 +1,84 @@
+
+We start off the week with a kickoff meeting on Monday morning. We go through the checklist with everybody and refine what's in it, making sure everybody has had input. Because we maintain so many packages, we have a spreadsheet where we keep track of the packages that are undergoing spring cleaning, and people can assign themselves to packages and mark them as completed when they're done.
+
+## Checklists, checklists, checklists
+
+We formalize these tasks into a checklist ([who doesn't love checklists](https://atulgawande.com/book/the-checklist-manifesto/)) via a function in usethis called `use_tidy_upkeep_issue()`. If you're a package developer and you use `use_release_issue()`, this will look familiar to you - it opens an issue in the package's GitHub repository with a checklist of tasks to guide you through what needs to be done. We update the function with the current year's checklist just prior to starting (and sometimes during) Spring Cleaning.
+
+Package maintainers then install the development version of usethis to get the current checklist, and call `usethis::use_tidy_upkeep_issue()` in their package to create the issue. If there are any tasks that aren't relevant to that particular repo it's easy to just edit the issue and remove it. To be really meta, here is the 2023 Spring Cleaning [upkeep issue for usethis](https://github.com/r-lib/usethis/issues/1791), created by usethis:
+
+```{r}
+#| echo: false
+#| fig-alt: Screenshot of an example upkeep issue in the usethis repository
+knitr::include_graphics("img/usethis-upkeep-issue.png")
+```
+
+We separated the tasks into "Necessary" and "Optional". The necessary tasks were those we needed to complete for all of our packages, and also were simple enough that we could be sure we would able to complete them. The optional items were those that were nice to have, and/or would take longer to complete.
+
+As always, the work to address the upkeep issues is submitted via [pull requests](https://github.com/tidymodels/dials/pull/275) - we try to complete the work, review, and merge all within the week with the intention of closing the upkeep issue by Friday. Any checklist items not completed in Spring Cleaning week are converted to their own issues (see the example in the above screenshot).
+
+## Wrapup
+
+Finally, we end the week with a wrap-up meeting - we do a retrospective on what worked, what didn't, and what we would change for next time. We also try to have a little fun! I made a small R package called [chatrbox](https://github.com/ateucher/chatrbox) that uses [ChatGPT](https://openai.com/blog/chatgpt) to generate R-themed Spring Cleaning text snippets. And Tracy Teal used [quarto](https://quarto.org/) to make certificates of achievement for each of us, complete with inspirational messages made with chatrbox!
```

Comment:
Maybe mention one change that we came up with this year — converting some of the more time consuming tasks into issues (i.e. from "use latest error handling conventions" to "make an issue to use error handing conventions"). This makes it more likely that we can cleanly complete the full checklist, meaning that there are lingering tasks to re-consider next year.